core/src/test/scala/org/apache/spark/HeartbeatReceiverSuite.scala

```diff
@@ -220,7 +222,7 @@ class HeartbeatReceiverSuite
     } else {
       assert(!response.reregisterBlockManager)
       // Additionally verify that the scheduler callback is called with the correct parameters
-      verify(scheduler).executorHeartbeatReceived(
+      verify(scheduler, VerificationModeFactory.atLeast(1)).executorHeartbeatReceived(
```

Comment:
@tedyu this still needs attention. Is the rest OK to commit @andrewor14 or are you interested in refining the tests?